internal/server/singleprocess/service_ui_deploy.go

```diff
@@ -0,0 +1,64 @@
+package singleprocess
+
+import (
+	"context"
+
+	"github.com/hashicorp/waypoint/internal/server/singleprocess/state"
+
+	pb "github.com/hashicorp/waypoint/internal/server/gen"
+)
+
+// TODO: test
+func (s *service) UI_ListDeployments(
+	cxt context.Context,
+	req *pb.UI_ListDeploymentsRequest,
+) (*pb.UI_ListDeploymentsResponse, error) {
+	deployList, err := s.state.DeploymentList(req.Application,
+		state.ListWithOrder(req.Order),
+		state.ListWithWorkspace(req.Workspace),
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	// TODO: make this more efficient. We should be able to just
+	// grab the relevant status report in one go, not have to iterate
+	// over all of them.
```

Comment:
You could enhance this to say we need to redo how `GetLatestStatusReport` is implemented. Right now it just calls its inherited func from app operation to get the latest item in the database. For us to target a deployment or release status report we'll have to not use that abstraction and instead write our own query for grabbing a status report if a target is requested.